File: 0118/solution.py
class Solution:
    def generate(self, numRows: int) -> list[list[int]]:
        res = [[1]]
        for row in range(1, numRows):
            res.append([1])
            for col in range(len(res[-2]) - 1):
                res[-1].append(res[-2][col] + res[-2][col + 1])
            res[-1].append(1)
        return res

File: 0118/solution.go
package main

func generate(numRows int) [][]int {
	res := [][]int{{1}}
	for i := 1; i < numRows; i++ {
		row := make([]int, i+1)
		row[0] = 1
		for j := 1; j < len(row)-1; j++ {
			row[j] = res[len(res)-1][j-1] + res[len(res)-1][j]
		}
		row[len(row)-1] = 1
		res = append(res, row)
	}
	return res
}